server/src/com/thoughtworks/go/server/service/AgentService.java

```diff
@@ -98,9 +97,16 @@ public void initialize() {
     }
 
     public void sync(Agents agents) {
+        syncWithDB(agents);
```

Comment:
This would be super expensive. `AgentService.sync` is invoked by `AgentChangeListener.onEntityConfigChange` which is invoked everytime an agent's config changes. Imagine in case of build.gocd setup itself when we bring up and tear down many hundred elastic agents each day. Each of those operations is going to sync every configured agent with the db. Is that really required? The other thing being, that callback is still part of config save operation, which means any other config save operation would be in queue until the db sync finishes off.